### DISCUSSION TITLE
Standardize architecture names within the AppImage community

### DISCUSSION THREAD

**Discussion by TheAssassin** [AppImage/AppImageKit#1299] [Ideas]
Given the fact that the architecture names vary a lot on the various distributions, among tools and even within the community, we should invest some effort into standardizing them.

The core of AppImage, namely appimagetool and the runtime, established a scheme already. The following table contains an incomplete listing of different schemes for comparison:

| Distribution | AMD64 | IA-32 | 64-bit ARM | 32-bit ARM |
| ------ | ----- | ----- | ----- | ----- | 
| AppImage core | x86_64 | i686 | aarch64 | armhf |
| Debian and derivatives | amd64 | i386 | aarch64 | armhf |
| RedHat and SUSE | x86_64 | i686 | arm64 |armhfp|

As you can see, AppImage is basically a mix of all of them. I personally like (and prefer) the Debian scheme. But since AppImage already kind of established a scheme, I'd suggest to implement it across the community.

The linuxdeploy projects currently use the AppImage core scheme with a single exception, IA-32 is referred to as `i386`. I will change that soon.

We should also document these names in the reference manual on https://docs.appimage.org.

**Comment by probonopd** [comment]
This is a big mess, not just within AppImage but overall.

Everyone seems to have an opinion on this. My 2 cents: "Normal users" with an "Intel" sticker on their computer would not recognize "amd" as something they own. If you ask me, I'd call it "x86-64bit", "x86-32bit", "arm-64bit", "arm-32bit". But I'm not proposing to invent yet-another names.

**Reply by TheAssassin** [reply]
Well, there are more differences. Technically, there is ARMv6 and ARMv7, both 32-bit. I think the first Raspberry Pi was ARMv6, actually. So there is a point in being more specific than "64 bit" (something that bit Microsoft, actually, because they always called their stuff 64-bit or x64, but given the fact that there are other ISAs like ARM as well that feature 64-bit and 32-bit, it was not such a good idea to use these simplifications after all).

My point is, I'd like to have some reference. And given the fact that the resulting AppImages' names is often autogenerated by appimagetool which already has its own concept, I'd like to make this the default.

Feel free to collect some more names and update the table above.

**Reply by probonopd** [reply]
True. There are also different "Intel 32-bit": 686,... 686. 
Then there are is little endian (armel) and big endian.
And the various ARM versions. (Which version is "armhf"? I don't know.)
This is complicated indeed.
